orangecontrib/spectroscopy/widgets/owcos.py

```diff
@@ -41,11 +41,9 @@ def calc_cos(table1, table2):
     sync = series1.T @ series2 / (len(series1) - 1)
 
     # Hilbert-Noda transformation matrix
-    hn = np.zeros((len(series1), len(series1)))
-    for i in range(len(series1)):
-        for j in range(len(series1)):
-            if i != j:
-                hn[i, j] = 1 / np.pi / (j - i)
+    i, j = np.ogrid[:len(series1), :len(series1)]
+    with np.errstate(divide='ignore'):
```

Comment:
Do not do these error state ignore because they are global (=can create problems if running multiple threads).

Reply:
I thought it ignores it only for the calculation in the "with" statement...?

Reply:
That would be correct if the assumption that only one thread is running holds. But this is not true in general...

Reply:
So, then we just swallow the errors? Or is it possible to catch them in another way?

Reply:
Swallow. :(

Reply:
Apparently starting from numpy 2.0 this would be [thread-safe](https://numpy.org/doc/stable/reference/generated/numpy.errstate.html)... Maybe when the numpy restriction in Orange is lifted we can revisit this. For now I'll remove it and make a note of this in the code.